lib/fs/basicfs_xattr_bsdish.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2022 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//go:build freebsd || netbsd
+// +build freebsd netbsd
+
+package fs
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+func listXattr(path string) ([]string, error) {
+	var attrs []string
+
+	// List the two namespaces explicitly and prefix any results with the
+	// namespace name.
+	namespacePrefixes := [...]string{unix.EXTATTR_NAMESPACE_USER: "user.", unix.EXTATTR_NAMESPACE_SYSTEM: "system."}
+	for _, nsid := range [...]int{unix.EXTATTR_NAMESPACE_USER, unix.EXTATTR_NAMESPACE_SYSTEM} {
+		buf := make([]byte, 1024)
+		size, err := unixLlistxattr(path, buf, nsid)
+		if errors.Is(err, unix.ERANGE) {
+			// Buffer is too small. Try again with a zero sized buffer to get
+			// the size, then allocate a buffer of the correct size.
+			size, err = unixLlistxattr(path, nil, nsid)
+			if err != nil {
+				return nil, fmt.Errorf("Listxattr %q: %w", path, err)
+			}
+			if size > len(buf) {
+				buf = make([]byte, size)
+			}
+			size, err = unixLlistxattr(path, buf, nsid)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Listxattr %q: %w", path, err)
```

Comment:
I dislike `%q` for logging/errors - an escaped path just looks bad in my opinion. What's the motivation for using it?

Reply:
... I don't recall, and I agree. Will change.